### ISSUE TITLE
Accepting contributions?

### ISSUE THREAD

**Issue by leighmcculloch** [davecgh/go-xdr#17]
Hi @davecgh,

Are you still maintaining this package and would you be open to accepting a contribution that adds an `xdr3` package, that is a copy of `xdr2` but adds optionals, unions, and change how pointers work to support using optionals?

We've been using a modified version of this package here: https://github.com/stellar/go-xdr and if you'd be up for us bringing our changes upstream we'd be up for preparing a PR.

This is approximately what the diff would look like, but we'd also need to merge downstream some of the changes in your master branch before we opened, so there would be some differences to keep xdr3 as a copy of xdr2 at the point in time the PR is opened:
https://github.com/davecgh/go-xdr/compare/master...stellar:master

No worries if you're no longer maintaining go-xdr, and if that is the case we'll move forward with developing our fork of go-xdr at https://github.com/stellar/go-xdr.

Cheers,
Leigh

cc @bartekn @ire-and-curses @marcopeereboom @rasky

**Comment by marcopeereboom** [comment]
If you are doing an xdr3 we should talk about memory allocation sizes for individual decode functions as well. Currently, we can set a global max size but it would be nice if we could pass a reasonably max size to individual decoders.

I'll help and I am sure we can coerce @davecgh to play along.

**Comment by leighmcculloch** [comment]
@marcopeereboom Does stellar/go-xdr@a1a70e705f65011b030182e92687858ada904e08 address the memory allocation?

**Comment by marcopeereboom** [comment]
I really like that.

**Comment by leighmcculloch** [comment]
@marcopeereboom I forgot to post back here on October 24th when I created the stellar/go-xdr#4 issue.

When I opened this issue we were debating whether to accept a contribution stellar/go-xdr#3 that made all the import paths canonical inside stellar/go-xdr so it could standalone and not be used as an alternative source of this repositories import path.

We decided to merge the contribution because @davecgh didn't respond here. We'll continue to develop the `xdr3` package over at `stellar/go-xdr`.

Thanks!
